### PR TITLE
Filter by methods

### PR DESCRIPTION
**Pull request by Mte90**
Start from https://github.com/perftools/xhgui/pull/237 and rewritten to add a new support.
Now other match the exact method/function support regular expression like from `Composer*` so all the method/function that fit this expression are removed.

Looking at the code now to filter the data is required to `/run/view?id=5cab781f51dc5f34fa735812&filter` to add a filter parameter in the URL or this code will be not executed.

### CODE REVIEW

File: src/Xhgui/Profile.php [hunk 1]
```diff
@@ -398,6 +398,49 @@ public function sort($dimension, $data)
...
+    }
+
+    /**
+     * @param $name
+     *
+     * @return bool|string
+     */
+    protected function getNameToFilter($name)
+    {
+        $length = strpos($name, '\\', 1);
```
Comment (markstory) [inline src/Xhgui/Profile.php]: Is this a `$length`? It looks like you're looking at the first namespace.

File: src/Xhgui/Profile.php [hunk 2]
```diff
@@ -398,6 +398,49 @@ public function sort($dimension, $data)
...
+    {
+        $filter = function ($item, $key) use ($filters) {
+            $nameToFilter = $this->getNameToFilter($key);
+            return !in_array($nameToFilter, $filters);
+        };
+        
+        $profileData = array_filter($profileData, $filter, ARRAY_FILTER_USE_BOTH);
+        foreach ($filters as $key => $item) {
+            foreach ($profileData as $key_item => $method) {
+                if (fnmatch($item, $key_item)) {
```
Comment (markstory) [inline src/Xhgui/Profile.php]: Variables are generally using camelBacked casing in xhgui.

File: src/Xhgui/Profile.php [hunk 3]
```diff
@@ -398,6 +398,49 @@ public function sort($dimension, $data)
...
+     * @param array $filters
+     *
+     * @return array
+     */
+    public function filter($profileData, $filters = [])
+    {
+        $filter = function ($item, $key) use ($filters) {
+            $nameToFilter = $this->getNameToFilter($key);
+            return !in_array($nameToFilter, $filters);
+        };
```
Comment (markstory) [inline src/Xhgui/Profile.php]: If you `fnmatch` the profile data below, why do you also need to pre-filter here?

Comment (Mte90) [reply src/Xhgui/Profile.php]: I was thinking the same, today I mixed the new feature with the previous one but yes can be optimized.

**Comment by Mte90** [comment]
For tests I have to prepare a dev environment but I will do them :-)

File: src/Xhgui/Profile.php (hunk 2 above)
Comment (Mte90) [reply src/Xhgui/Profile.php]: sorry, I am used the snakeCase because I work a lot on WordPress.

**Comment by Mte90** [comment]
Started working on tests but not yet finished, required a bit with issue on docker on my system.

**Comment by Mte90** [comment]
£ new unit tests added :-)
Let me know if it is missing something!

**Pull request merged**